smac/epm/base_rf.py

```diff
@@ -6,19 +8,35 @@
     UniformFloatHyperparameter,
     UniformIntegerHyperparameter,
     Constant,
+    ConfigurationSpace,
 )
 
 
 class BaseModel(AbstractEPM):
 
-    def __init__(self, configspace, types, bounds, seed, **kwargs):
+    def __init__(
+        self,
+        configspace: ConfigurationSpace,
+        types: np.ndarray,
+        bounds: np.ndarray,
+        seed: int,
+        instance_features: Optional[np.ndarray] = None,
+        pca_components: Optional[int] = None,
+    ) -> None:
         """
         Abstract base class for all random forest models.
         """
-        super().__init__(configspace=configspace, types=types, bounds=bounds, seed=seed, **kwargs)
+        super().__init__(
+            configspace=configspace,
+            types=types,
+            bounds=bounds,
+            seed=seed,
+            instance_features=instance_features,
+            pca_components=pca_components,
+        )
 
         self.rng = np.random.RandomState(seed)
-        self.impute_values = dict()
+        self.impute_values = dict()  # type: Dict[int, Optional[float]]
```

Comment:
In case a hyperparameter is not conditional there is nothing to impute. To denote this information, we add a None here. Should I add another check ~line 57 to not impute anything if impute_values[idx] is sot to None, or should there maybe be an indicator array with boolean entries?

Reply:
I see.
I like the idea with an indicator array best.